src/Modal.js

```diff
@@ -277,6 +277,11 @@ const Modal = React.createClass({
     }
   },
 
+  componentWillUnmount() {
+    this.onHide();
+    this.props.onHide();
```

Comment:
I can appreciate that though I disagree @taion perhaps can also weigh in.

regardless of that point right now the PR needs to add a guard that checks that the modal is even open before calling anything, could you make that change?